pkg/tickers/retry.go

```diff
@@ -0,0 +1,84 @@
+package tickers
+
+import (
+	"context"
+	"time"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+// UpkeepPayload represents the payload for an upkeep task.
+//type UpkeepPayload struct {
+//	Hash string // Hash uniquely identifies the upkeep payload
+//}
+
+// CheckResult represents the result of a check pipeline.
+type CheckResult struct {
+	Payload   ocr2keepers.UpkeepPayload
+	Retryable bool
+	// ... other properties
+}
+
+// RetryTicker represents a ticker that handles retries.
+type RetryTicker struct {
+	*TimeTicker
+	nextRetries  map[time.Time]ocr2keepers.UpkeepPayload
+	newDurations map[string]time.Duration
+}
+
+// Retry adds a retryable result to the RetryTicker.
+func (rt *RetryTicker) Retry(result CheckResult) error {
+	payload := result.Payload
+	retryable := result.Retryable
+
+	if retryable {
+		newDuration, ok := rt.newDurations[payload.Hash]
+		if !ok {
+			newDuration = 1 * time.Second // First time for this payload, set initial duration
+		}
+
+		nextRunTime := time.Now().Add(newDuration)
+		rt.nextRetries[nextRunTime] = payload
+
+		newDuration *= 2 // Double the duration for the next retry
+		rt.newDurations[payload.Hash] = newDuration
+	}
+
+	return nil
+}
+
+// getterFn is a function that retrieves the retryTick for the given time.
+func (rt *RetryTicker) getterFn(ctx context.Context, t time.Time) (Tick, error) {
+	upkeepPayloads := []ocr2keepers.UpkeepPayload{}
+	for key, value := range rt.nextRetries {
+		if t.After(key) {
+			upkeepPayloads = append(upkeepPayloads, value)
+			delete(rt.nextRetries, key)
```

Comment:
Are we removing this too early? What if the retry is not successful?

As discussed with @ferglor, we need to track the retry status somehow, and mark the Payload as retried_and_succeeded if it went through. 

@EasterTheBunny can shed some light here.